lib/camunda-cloud/ElementTemplatesReplaceProvider.js

```diff
@@ -0,0 +1,105 @@
+import {
+  getBusinessObject,
+  isAny
+} from 'bpmn-js/lib/util/ModelUtil';
+
+/**
+ * A replace menu provider that allows to replace elements with
+ * element templates.
+ */
+function ElementTemplatesReplaceProvider(popupMenu, translate, elementTemplates) {
+
+  this._popupMenu = popupMenu;
+  this._translate = translate;
+  this._elementTemplates = elementTemplates;
+
+  this.register();
+}
+
+ElementTemplatesReplaceProvider.$inject = [
+  'popupMenu',
+  'translate',
+  'elementTemplates'
+];
+
+
+/**
+ * Register replace menu provider in the popup menu
+ */
+ElementTemplatesReplaceProvider.prototype.register = function() {
+  this._popupMenu.registerProvider('bpmn-replace', this);
+};
+
+/**
+ * Get all element templates that can be used to replace the given element.
+ *
+ * @param {djs.model.Base} element
+ *
+ * @return {Array<Object>} a list of element templates as menu entries
+ */
+ElementTemplatesReplaceProvider.prototype.getEntries = function(element) {
+
+  const templates = this._getMatchingTemplates(element);
+  return templates.map(template => {
+
+    const {
+      id,
+      icon = {},
+    } = template;
+
+    const group = {
+      id: 'templates',
+      name: this._translate('Templates')
+    };
+
+    return {
```

Comment:
We'd need to make sure that a template can provide it's own group (overriding the default `Templtae` group).

Reply:
They allow groups for elements, referred to as "category", as an inofficial extension (not part of the schema yet, but actively used in web-modeler, as well as our connectors extension example, cf. https://github.com/bpmn-io/bpmn-js-connectors-extension/blob/master/src/replace-menu/ReplaceMenu.js#L64, https://github.com/camunda/web-modeler/blob/main/webapp/src/App/Pages/Diagram/BpmnJSExtensions/connectorsExtension/.camunda/element-templates/aws-lambda-connector.json#L8).

Reply:
We could consider making this an official extension point (cf. https://github.com/bpmn-io/element-templates/issues/8).

Reply:
Transforming is not a big deal, right? The reason we don't use "group" in element templates is simply that group is already used (to group template entries).

I'd keep things as is (transform), and ensure that mid-term elements are actually grouped at the core (https://github.com/bpmn-io/diagram-js/issues/713).

Reply:
Of course, before we blog about the new bpmn-js release everything is possible... :shushing_face:.